internal/feature_test.go

```diff
@@ -193,69 +194,224 @@ func testDisconnectWhileRunningOperation(feature *ScriptBasedSoftwareUpdatable,
 	waitDisconnect.Wait()
 	defer connectFeature(t, mc, feature, getDefaultFlagValue(t, flagFeatureID))
 	if install {
-		checkInstallStatusEvents(statuses, t)
+		checkInstallStatusEvents(0, statuses, t)
 	} else {
-		checkDownloadStatusEvents(statuses, t)
+		checkDownloadStatusEvents(0, statuses, t)
 	}
 }
 
+// TestScriptBasedDownloadAndInstallMixedResources tests ScriptBasedSoftwareUpdatable core functionality: init, install and download operations,
+// working with both downloadable and local resources
+func TestScriptBasedDownloadAndInstallMixedResources(t *testing.T) {
+	dataDir := "tempdata"
+	assertDirs(t, dataDir, true)
+	defer os.RemoveAll(dataDir)
+
+	// Prepare
+	storageDir := assertDirs(t, testDirFeature, false)
+	// Remove temporary directory at the end.
+	defer os.RemoveAll(storageDir)
+
+	feature, mc, err := mockScriptBasedSoftwareUpdatable(t, &testConfig{
+		clientConnected: true, featureID: getDefaultFlagValue(t, flagFeatureID), storageLocation: storageDir, mode: modeLax,
+	})
+	if err != nil {
+		t.Fatalf("failed to initialize ScriptBasedSoftwareUpdatable: %v", err)
+	}
+	defer feature.Disconnect(true)
+
+	a, aBody := "a.txt", "test"
+	aPath, aHash := createLocalArtifact(t, dataDir, a, aBody)
+	b, bBody := "b.txt", "test"
+	bPath, bHash := createLocalArtifact(t, dataDir, b, bBody)
+
+	// Prepare/Close simple HTTP server used to host testing artifacts
+	w := storage.NewTestHTTPServer(testDefaultHost, "", 0, t)
+	w.Host(true, false, "", "")
+	w.AddInstallScript()
+	defer w.Close()
+
+	installScript := w.GenerateSoftwareArtifacts(false, "install")
+	artifacts := []*hawkbit.SoftwareArtifactAction{
+		convertLocalArtifact(aPath, a, aHash, len(aBody)),
+		convertLocalArtifact(bPath, b, bHash, len(bBody)),
+		installScript[0],
+	}
+
+	testDownloadInstall(feature, mc, artifacts, true, b, t)
+}
+
+// TestScriptBasedDownloadAndInstallLocalResources tests ScriptBasedSoftwareUpdatable core functionality: init, install and download operations,
+// but working with local resources
+func TestScriptBasedDownloadAndInstallLocalResources(t *testing.T) {
+	tempDir := "tempdata"
```

Comment:
- usually, `tmp` is used, not `temp` - for example: /var/tmp
- tempdata - looks like runtime data, `test` can be included to be more accurate